akka-sample-kafka-to-sharding-scala/processor/src/main/scala/sample/sharding/kafka/UserEventsKafkaProcessor.scala

```diff
@@ -89,5 +106,125 @@ object UserEventsKafkaProcessor {
       }
       .narrow
   }
+}
+
+object ShardingToReliableDelivery {
+  def sink[M](
+               producerController: akka.actor.typed.ActorRef[ShardingProducerController.Command[M]]
+             ): Sink[ShardingEnvelope[M], Future[Done]] = {
+    Sink.fromGraph(new ShardingToReliableDeliveryStage(producerController))
+  }
+}
+
+/**
+ * Forwards messages to sharded entities via reliable delivery.
+ *
+ * Will keep requesting messages until there is a max number of messages buffered or
+ * a single entity has too many buffered messages
+ *
+ * Should one slow entity be allowed to slow everything down? Option to start dropping for a given entity?
+ */
+class ShardingToReliableDeliveryStage[M](
```

Comment:
This seemed nicer than having an actor that managed interaction with reliable delivery and asking from the kafka stream. We could clean up and add to reliable delivery